Do not prepend implicit type selectors to :host rules

Ahmad-S792 opened the pull request:
#### f92a593eb387053331584c5fd709227ce52f6851
<pre>
Do not prepend implicit type selectors to :host rules

<a href="https://bugs.webkit.org/show_bug.cgi?id=262907">https://bugs.webkit.org/show_bug.cgi?id=262907</a>

Reviewed by Tim Nguyen.

Merge: <a href="https://chromium-review.googlesource.com/c/chromium/src/+/518019">https://chromium-review.googlesource.com/c/chromium/src/+/518019</a>

Web Specification: <a href="https://drafts.csswg.org/selectors-4/#featureless">https://drafts.csswg.org/selectors-4/#featureless</a>

&quot;If a given selector is allowed to match a featureless element, it must do
so while ignoring the default namespace.&quot;

:host should not have implicit type selectors pre-pended for
universal rules with default `@namespace`. Default @namespace`
applies to type and universal selectors inside :host(), but a
default `@namespace` rule should not affect :host.
That is, :host matches the host element regardless of any `@namespace`
rules.

* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(CSSSelectorParser::prependTypeSelectorIfNeeded):
* LayoutTests/TestExpectations: Removed now &apos;failing&apos; test

Canonical link: <a href="https://commits.webkit.org/269134@main">https://commits.webkit.org/269134@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c21597eb8a313e85e92a564ad0f67f0038f591b3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21690 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/21962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22734 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23554 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/20081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21932 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/26174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22220 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21917 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/26174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/18784 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24408 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/26174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/19637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/26174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/19848 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/23792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/20334 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19662 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23887 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2686 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20253 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Can you reference: https://drafts.csswg.org/selectors-4/#featureless in the commit message?

Ahmad-S792 committed:
Do not prepend implicit type selectors to :host rules

https://bugs.webkit.org/show_bug.cgi?id=262907

Reviewed by Tim Nguyen.

Merge: https://chromium-review.googlesource.com/c/chromium/src/+/518019

Web Specification: https://drafts.csswg.org/selectors-4/#featureless

"If a given selector is allowed to match a featureless element, it must do
so while ignoring the default namespace."

:host should not have implicit type selectors pre-pended for
universal rules with default `@namespace`. Default @namespace`
applies to type and universal selectors inside :host(), but a
default `@namespace` rule should not affect :host.
That is, :host matches the host element regardless of any `@namespace`
rules.

* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(CSSSelectorParser::prependTypeSelectorIfNeeded):
* LayoutTests/TestExpectations: Removed now 'failing' test

Canonical link: https://commits.webkit.org/269134@main
